### PR TITLE
React Native implementation of Search and Search UI

### PR DESCRIPTION
**Pull request by chrmod**
Additional TODO:

- [ ] filter abi correctly (now we ship arm and x86)

### CODE REVIEW

File: mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import org.json.JSONObject;
+
+import java.util.Iterator;
+
+/**
+ * @author Khaled Tantawy
+ */
+public class Utils {
+    static WritableMap convertJsonToMap(JSONObject jsonObject) throws JSONException {
+        WritableMap map = new WritableNativeMap();
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java]: final

File: mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+        Iterator<String> iterator = jsonObject.keys();
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java]: here also and in other methods as well

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java [hunk 3]
```diff
@@ -870,7 +880,7 @@ public boolean onInterceptMotionEvent(View view, MotionEvent event) {
             "Search:OpenLink",
             "Privacy:Count",
             "Search:Idle",
-            "Search:Ready",
+            "Search:renderResults",
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java]: Naming style not consistent

**Comment by ravjit-cliqz** [comment]
There are extra line breaks in multiple places. Remove those. 
@spacifici should we add `Copyright Cliqz` to the files which are in the cliqz/freshtab package?

**Comment by chrmod** [comment]
@ravjit-cliqz @sharath-cliqz thanks for review!

**Comment by giota-cliqz** [comment]
@chrmod I removed the WIP label. Is there anything else missing from your side?

File: mozilla-release/mobile/android/base/java/com/cliqz/react/BridgePackage.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        final List<ViewManager> viewManagers = new ArrayList<>();
+        viewManagers.add(new NativeDrawableManager());
+
+        return viewManagers;
+    }
+
+    public class Bridge extends ReactContextBaseJavaModule implements BundleEventListener {
+        private final ReactApplicationContext mReactContext;
+
+
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/com/cliqz/react/BridgePackage.java]: extra line

File: mozilla-release/mobile/android/base/java/com/cliqz/react/BridgePackage.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        @Override
+        public void handleMessage(String event, GeckoBundle message, EventCallback callback) {
+            switch (event) {
+                case "Search:Search":
+                    final String query = GeckoBundleUtils.safeGetString(message, "q");
+                    mReactContext
+                            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                            .emit("search:search", query);
+                    break;
+
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/com/cliqz/react/BridgePackage.java]: extra line

File: mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                map.putString(key, (String) value);
+            } else {
+                map.putString(key, value.toString());
+            }
+        }
+        return map;
+    }
+
+    static WritableArray convertJsonToArray(JSONArray jsonArray) throws JSONException {
+        WritableArray array = new WritableNativeArray();
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java]: final

File: mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+        for (int i = 0; i < jsonArray.length(); i++) {
+            Object value = jsonArray.get(i);
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java]: final

File: mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                array.pushString((String) value);
+            } else {
+                array.pushString(value.toString());
+            }
+        }
+        return array;
+    }
+
+    static JSONObject convertMapToJson(ReadableMap readableMap) throws JSONException {
+        JSONObject object = new JSONObject();
```
Comment (ravjit-cliqz) [inline mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java]: final

Comment (ravjit-cliqz) [reply mozilla-release/mobile/android/base/java/com/cliqz/react/Utils.java]: and also at other places in this class

File: mozilla-release/mobile/android/base/java/com/cliqz/react/SearchUI.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+import org.mozilla.gecko.preferences.PreferenceManager;
+import org.mozilla.gecko.util.AppBackgroundManager;
+import org.mozilla.geckoview.BuildConfig;
+
+/**
+ * @author Khaled Tantawy
+ */
+public class SearchUI implements ReactInstanceManager.ReactInstanceEventListener {
+
+
```
Comment (jayeshsolanki93) [inline mozilla-release/mobile/android/base/java/com/cliqz/react/SearchUI.java]: random whitespace

File: mozilla-release/mobile/android/base/java/com/cliqz/react/SearchUI.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+            mReactRootView.unmountReactApplication();
+        }
+    }
+
+    @Override
+    public void onReactContextInitialized(ReactContext context) {
+        mReactContext = context;
+    }
+
+
```
Comment (jayeshsolanki93) [inline mozilla-release/mobile/android/base/java/com/cliqz/react/SearchUI.java]: random whitespace

**Pull request merged**